internal/hns/hnsnetwork.go

```diff
@@ -98,6 +98,12 @@ func (network *HNSNetwork) Create() (*HNSNetwork, error) {
 	title := "hcsshim::HNSNetwork::" + operation
 	logrus.Debugf(title+" id=%s", network.Id)
 
+	for _, subnet := range network.Subnets {
+		if (subnet.AddressPrefix != "") && (subnet.GatewayAddress == "") {
+			return nil, fmt.Errorf("Network Create Error, subnet has address prefix but no gateway specified.")
```

Comment:
Errors in Go should start with a lowercase letter (unless the first word is a proper noun, of course) and should not end with punctuation. That makes them easier to wrap.

Also, you don't need `fmt.Errorf` here since there are no format substitutions. You should use `errors.New` in this case.

Ditto elsewhere in this PR.